Source/WebCore/loader/CrossOriginAccessControl.cpp

```diff
@@ -122,6 +122,21 @@ ResourceRequest createAccessControlPreflightRequest(const ResourceRequest& reque
             preflightRequest.setHTTPHeaderField(HTTPHeaderName::AccessControlRequestHeaders, headerBuffer.toString());
     }
 
+    if (includeFetchMetadata) {
+        auto requestOrigin = SecurityOrigin::create(request.url());
+        if (requestOrigin->isPotentiallyTrustworthy()) {
+            preflightRequest.setHTTPHeaderField(HTTPHeaderName::SecFetchMode, "cors"_s);
+            preflightRequest.setHTTPHeaderField(HTTPHeaderName::SecFetchDest, "empty"_s);
+
+            if (securityOrigin.isSameOriginAs(requestOrigin))
+                preflightRequest.addHTTPHeaderField(HTTPHeaderName::SecFetchSite, "same-origin"_s);
+            else if (securityOrigin.isSameSiteAs(requestOrigin))
+                preflightRequest.addHTTPHeaderField(HTTPHeaderName::SecFetchSite, "same-site"_s);
+            else
+                preflightRequest.addHTTPHeaderField(HTTPHeaderName::SecFetchSite, "cross-site"_s);
```

Comment:
SecFetchSite should be the same between original request and preflight.
includeFetchMetadata could be computed from the original request, say hasHTTPHeaderField(HTTPHeaderName::SecFetchSite).

Reply:
I like the idea but this header isn't set on the regular request at this point.

Reply:
It should be set for WK2 since preflight should only be done from network process, not web content.
It is not set for WK1, where this code could be used.

Reply:
I'm only testing it against WK2. The headers just aren't set.